demo/web/src/main/notebooks/Python_AI/01 Deephaven with SciKit-Learn.md

```diff
@@ -0,0 +1,189 @@
+    outputs = [learn.Output("PredictedClass", numpy_to_table_integer, "int")],
+    batch_size = iris_test.size()
+)
+```
+
+The K-Nearest neighbors classifier worked great!  We just did some classifications using a simple model on a static data set.  That's kind of cool, but this data set is neither large or real-time.
+
+## Define bounds for making realistic faux Iris measurements
+
+Let's create a real-time table of faux Iris measurements and apply our fitted classifier.
+We need to make sure our faux measurements are realistic, so let's grab the minimum and maximum observation values and use those to generate random numbers.
+
+```python
+min_petal_length, max_petal_length = raw_iris["PetalLengthCM"].min(), raw_iris["PetalLengthCM"].max()
+min_petal_width, max_petal_width = raw_iris["PetalWidthCM"].min(), raw_iris["PetalWidthCM"].max()
+min_sepal_length, max_sepal_length = raw_iris["SepalLengthCM"].min(), raw_iris["SepalLengthCM"].max()
+min_sepal_width, max_sepal_width = raw_iris["SepalWidthCM"].min(), raw_iris["SepalWidthCM"].max()
+```
+
+## Write the faux measurements to a table in real-time
+
+With these quantities now calculated and stored in memory, we need to set up alive table that we can write measurements to.  To keep it simple, we'll write measurements to the table once per second for a minute.
+
+```python
+table_writer = DynamicTableWriter(
+    ["SepalLengthCM", "SepalWidthCM", "PetalLengthCM", "PetalWidthCM"],
+    [dht.double] * 4
+)
+
+live_iris = table_writer.getTable()
+
+def write_faux_iris_measurements():
+    for i in range(60):
+        sepal_length = np.round(random.uniform(min_sepal_length, max_sepal_length), 1)
+        sepal_width = np.round(random.uniform(min_sepal_width, max_sepal_width), 1)
+        petal_length = np.round(random.uniform(min_petal_length, max_petal_length), 1)
+        petal_width = np.round(random.uniform(min_petal_width, max_petal_width), 1)
+
+        table_writer.logRow(sepal_length, sepal_width, petal_length, petal_width)
+        time.sleep(1)
+
+thread = threading.Thread(target = write_faux_iris_measurements)
+thread.start()
+```
+
+## Use the trained model on the live data
+
+Now we've got some faux live incoming measurements.  We can just use our fitted K-Neighbors classifier on the live table!
+
+```python
+iris_classifications_live = learn.learn(
+    table = live_iris, 
+    model_func = use_fitted_knn,
+    inputs = [learn.Input(["SepalLengthCM", "SepalWidthCM", "PetalLengthCM", "PetalWidthCM"], table_to_tensor_double)],
+    outputs = [learn.Output("PredictedClass", tensor_to_table_integer, "int")],
+    batch_size = 5
+)
+```
+
+And there we have it.  Our classifier is working on live data.  The only extra work we needed to make that happen was to set up the real-time data stream.  Pretty simple!  This may be a toy problem, but the steps hold true to more complex ones.
```

Comment:
```suggestion
And there we have it.  Our classifier is working on live data.  The only extra work required was to set up the real-time data stream.  Pretty simple!  This may be a toy problem, but the steps hold true to more complex ones.
```

Reply:
should change above, too